dotCMS/src/main/java/com/dotmarketing/portlets/structure/ajax/FieldVariableAjax.java

```diff
@@ -108,6 +116,10 @@ public String deleteFieldVariable(String fieldVarId) throws DotDataException, Do
 		FieldVariable fieldVar = fieldAPI.findFieldVariable(fieldVarId, user, respectFrontendRoles);
 		fieldAPI.deleteFieldVariable(fieldVar, user, respectFrontendRoles);
 		
+		Field field = APILocator.getContentTypeFieldAPI().find(fieldVar.getFieldId());
+		ContentTypeAPI tapi = APILocator.getContentTypeAPI(user);

```

Comment:
it seems you were using the same API before, if it is the same user you can ask for the instance before both uses